### PR TITLE
Swagger specification for Service Fabric REST Client APIs for version 5.6

### PR DESCRIPTION
**Pull request by vipul-modi**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - x ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by olydis** [comment]
related discussion: https://github.com/Azure/azure-rest-api-specs-pr/pull/70

**Comment by olydis** [comment]
>           "description": "The contents of the cluster manifest file."

just making sure: this is really an `object` with one `string` property? think I've seen the corresponding method just return a `string` in a previous version of the Swagger, can you confirm that you service returns this object instead?

---
Refers to: servicefabric/5.6/swagger/servicefabric.json:5078 in 41ac528. [](commit_id = 41ac528474ffc05d84cd71a5354f0a59691c5872, deletion_comment = False)

**Comment by olydis** [comment]
>     "ClusterManifest": {

only used as a response model. please mark all properties as `readOnly`.

---
Refers to: servicefabric/5.6/swagger/servicefabric.json:5072 in 41ac528. [](commit_id = 41ac528474ffc05d84cd71a5354f0a59691c5872, deletion_comment = False)

**Comment by olydis** [comment]
>     "ClusterHealth": {

only used as a response model. please mark all properties as readOnly.

Will omit this suggestion for all further occurrences. It applies to all models that are not used as (or are part of) parameters.

---
Refers to: servicefabric/5.6/swagger/servicefabric.json:4966 in 41ac528. [](commit_id = 41ac528474ffc05d84cd71a5354f0a59691c5872, deletion_comment = False)

**Comment by olydis** [comment]
>             "$ref": "#/parameters/ClusterHealthPoliciesOptionalBodyParam"

when this parameter is not specified, does this method behave any different from the previous one? if not, I'd make this parameter required or at least make it the first parameter. Otherwise, this method is somewhat confusing. Or is this mimicking your existing SDK and hence can't change?

---
Refers to: servicefabric/5.6/swagger/servicefabric.json:107 in 41ac528. [](commit_id = 41ac528474ffc05d84cd71a5354f0a59691c5872, deletion_comment = False)

**Comment by olydis** [comment]
>       "description": "Represents the health chunk of the cluster.\nContains the cluster aggregated health state, and the cluster entities that respect the input filter.\n",

there is an input filter?

---
Refers to: servicefabric/5.6/swagger/servicefabric.json:4995 in 41ac528. [](commit_id = 41ac528474ffc05d84cd71a5354f0a59691c5872, deletion_comment = False)

**Comment by olydis** [comment]
>     "ClusterUpgradeHealthPolicyObject": {

general comment: do you depend on the `Object` suffix already? usually we advice against these things

---
Refers to: servicefabric/5.6/swagger/servicefabric.json:8313 in 41ac528. [](commit_id = 41ac528474ffc05d84cd71a5354f0a59691c5872, deletion_comment = False)

**Comment by olydis** [comment]
>     "title": "Service Fabric Client APIs",

note that AutoRest derives the client class from the `title`. Currently this results in `ServiceFabricClientAPIs` which is not the best name for a class.

---
Refers to: servicefabric/5.6/swagger/servicefabric.json:4 in 41ac528. [](commit_id = 41ac528474ffc05d84cd71a5354f0a59691c5872, deletion_comment = False)

File: servicefabric/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Service Fabric REST Client API Specifications
```
Comment (olydis) [inline servicefabric/README.md]: very nice documentation and descriptions in general! :+1:

File: servicefabric/5.6/examples/GetPartitionRestartProgress.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  "operationId": "GetPartitionRestartProgress",
+  "description": "This shows an example of possible output when GetPartitionRestartProgress has reached a State of Completed.  If the State has a value of Running, only the State field will have a value.",
+  "parameters": {
+    "api-version": "3.0",
+    "serviceId": "fabric:/myapp/myservice",
+    "partitionId": "1daae3f5-7fd6-42e9-b1ba-8c05f873994d",
+    "OperationId": "6eb7f0d3-49ca-4cb1-81b7-ac2d56b5c5d1",
```
Comment (olydis) [inline servicefabric/5.6/examples/GetPartitionRestartProgress.json]: syntax error!

File: servicefabric/5.6/examples/GetQuorumLossProgress-Running.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+  "operationId": "GetQuorumLossProgress",
+  "description": "This shows an example of possible output when GetQuorumLossProgress has a State of Running.  If the State has a value of Running, only the State field will have a value.",
+  "parameters": {
+    "api-version": "3.0",
+    "serviceId": "fabric:/myapp/myservice",
+    "partitionId": "1daae3f5-7fd6-42e9-b1ba-8c05f873994d",
+    "OperationId": "81fbaee7-4f49-44bb-a350-4cb2b51dbf26",
```
Comment (olydis) [inline servicefabric/5.6/examples/GetQuorumLossProgress-Running.json]: syntax

File: servicefabric/5.6/examples/GetQuorumLossProgress.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "operationId": "GetQuorumLossProgress",
+  "description": "This shows an example of possible output when GetQuorumLossProgress has reached a State of Completed.  If the State has a value of Running, only the State field will have a value.",
+  "parameters": {
+    "api-version": "3.0",
+    "serviceId": "fabric:/myapp/myservice",
+    "partitionId": "1daae3f5-7fd6-42e9-b1ba-8c05f873994d",
+    "OperationId": "81fbaee7-4f49-44bb-a350-4cb2b51dbf26",
```
Comment (olydis) [inline servicefabric/5.6/examples/GetQuorumLossProgress.json]: syntax

File: servicefabric/5.6/examples/GetPartitionRestartProgress.json (hunk 2 above)
Comment (olydis) [inline servicefabric/5.6/examples/GetPartitionRestartProgress.json]: trailing `,`

Comment (VipulM-MSFT) [inline servicefabric/5.6/examples/GetPartitionRestartProgress.json]: Will fix

**Comment by VipulM-MSFT** [comment]
The readOnly property marking improvements are planned for the next release. The rest of the comments related to the naming and structure or due to the existing product. The ServiceFabricClientAPIs name for the generated client is ok for now and as it matches the current SDK.

**Comment by olydis** [comment]
Also, once again making sure you really rely on having no method groups, i.e. provide this flat experience:
image

We do have dogfoody ways to use the `tags` for grouping in case that is what your current SDK does. 🙂

**Comment by VipulM-MSFT** [comment]
Since we are not publishing the C# Swagger generated SDK right now. We can use the right naming convention on the next update or rely on the new generation feature.

**Comment by olydis** [comment]
> Since we are not publishing the C# Swagger generated SDK right now.

darn, of course... sounds good to me!

**Comment by lmazuel** [comment]
@olydis @VipulM-MSFT but the Python SDK is released (just sayin)
https://github.com/Azure/azure-sdk-for-python/pull/1147

FYI @samedder

This means that the future SDK will have breaking changes, and version should be dumped to 6.x (semantic versionning) since you hasked me to sync ApiVersion and package version. But probably you will not bump the ApiVersion, so we will have a problem to respect semantic versionning for the Python package :(

**Comment by olydis** [comment]
@lmazuel Thanks for pointing that out, this is indeed less than optimal!

**Comment by VipulM-MSFT** [comment]
The next update of the specification will have higher version. We are planning on updating the swagger for every release of the platform and it has different version number.

**Comment by samedder** [comment]
@VipulM-MSFT so what is the plan for the swagger version next time we have a breaking change? Is it going to match the Service Fabric product or follow semantic versioning?

**Comment by VipulM-MSFT** [comment]
@olydis - let us know if you have any other question. We are looking forward to getting this merged and published.

**Comment by olydis** [comment]
@VipulM-MSFT Model validation has found problems, see https://travis-ci.org/Azure/azure-rest-api-specs/jobs/229277650 (starting line 1116). At first sight it looks like either `ApplicationTypeStatus` is wrongly modeled as a `string` or your examples are wrong as they have an integer in that position.

**Comment by VipulM-MSFT** [comment]
@olydis - you are right - this changed on the last day from int to string. I will update the example. Why I could not see this failure?

**Comment by VipulM-MSFT** [comment]
@olydis - The validation tool is requiring that example have 204 response code. This is not possible. Based on the parameters passed in the request, you only get 200 response. The example should have only one request-response pair.

**Comment by olydis** [comment]
>       "type": "string",

type `string` with format `int64` is not a thing! Either remove the format or change the type to `integer`, depending on what your API actually does.

---
Refers to: servicefabric/5.6/swagger/servicefabric.json:10985 in 41ac528. [](commit_id = 41ac528474ffc05d84cd71a5354f0a59691c5872, deletion_comment = False)

**Comment by olydis** [comment]
>       "type": "string",

same as above

---
Refers to: servicefabric/5.6/swagger/servicefabric.json:10994 in 41ac528. [](commit_id = 41ac528474ffc05d84cd71a5354f0a59691c5872, deletion_comment = False)

**Comment by olydis** [comment]
>       "enum": [

capitalization of enum items seems to disagree with example `GetFaultOperationList`

---
Refers to: servicefabric/5.6/swagger/servicefabric.json:10326 in 41ac528. [](commit_id = 41ac528474ffc05d84cd71a5354f0a59691c5872, deletion_comment = False)

**Comment by olydis** [comment]
> Why I could not see this failure?

Because we currently allow CI to pass even if some checks fail as they are still in development.

**Review by olydis — APPROVED** [review]
LGTM

**Pull request merged**